addons/mail/static/tests/discuss_app/jump_to_present.test.js

```diff
@@ -234,3 +235,62 @@ test("Post message when seeing old message should jump to present", async () =>
         after: [".o-mail-Message-content", { text: "Most Recent!" }], // should load around present
     });
 });
+
+test("jump to present triggers with at least 10 long messages seen", async () => {
+    // when messages are short, 3 x PRESENT_VIEWPORT_THRESHOLD is used, otherwise
+    // this is 10 x PRESENT_MESSAGE_THRESHOLD
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({
+        channel_type: "channel",
+        name: "General",
+        channel_member_ids: [Command.create({ partner_id: serverState.partnerId })],
+    });
+    for (let i = 0; i < 100; i++) {
+        pyEnv["mail.message"].create({
+            body: "<p>Non Empty Body</p>".repeat(100),
+            message_type: "comment",
+            model: "discuss.channel",
+            res_id: channelId,
+        });
+    }
+    const newestMessageId = pyEnv["mail.message"].create({
+        body: "<p>Newest</p>",
+        message_type: "comment",
+        model: "discuss.channel",
+        res_id: channelId,
+    });
+    const [selfMember] = pyEnv["discuss.channel.member"].search_read([
+        ["partner_id", "=", serverState.partnerId],
+        ["channel_id", "=", channelId],
+    ]);
+    pyEnv["discuss.channel.member"].write([selfMember.id], {
+        new_message_separator: newestMessageId + 1,
+    });
```

Comment:
Same goes for those 6 lines; they are not needed, are they? I suggest removing them to streamline the test flow.

Reply:
They are necessary to ensure that initially we are at present. Otherwise we see oldest message.

The feature of `new_message_separator` is designed to need explicit value to last message to jump to present, otherwise it doesn't. IMO this is not a good default behaviour, and the complexity of the test shows it.

So I can't only substitute these LOCs by a step that jumps to present explicitly. This also feels unnatural and could even be prone to some other runbot race conditions. I prefer defining the data properly for the time being.

Reply:
Oh right, good point. Thanks you for the explanation :)